xfer/publisher.go

```diff
@@ -59,9 +59,7 @@ func (p HTTPPublisher) String() string {
 }
 
 // Publish publishes the report to the URL.
-func (p HTTPPublisher) Publish(buf *Buffer) error {
-	defer buf.Put()
-
+func (p HTTPPublisher) Publish(buf Buffer) error {
```

Comment:
If you're just using the io.Reader parts of the buffer, here, maybe you can have it take an io.Reader?